gems/aws-sdk-core/lib/aws-sdk-core/errors.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  module Errors
+    # Raised when ARN string input doesn't follow the standard:
+    # https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-arns
+    class InvalidARNError < RuntimeError; end
```

Comment:
This module may not be necessary anymore. Thoughts? I actually think maybe this error should be defined in the same file as the ARN parser or even the ARN class. I think a shared errors module is necessary only if multiple modules may raise an error (we could "promote" things this way?) - for now, only arn parser will ever raise this.

Reply:
Yeah, I think thats fair.  We can keep errors specific to a single module in that module.